dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/cdi/OpenTelemetryProducer.java

```diff
@@ -0,0 +1,161 @@
+
+import org.eclipse.microprofile.config.Config;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.exporter.jaeger.JaegerGrpcSpanExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder;
+
+public class OpenTelemetryProducer {
+
+    private String instrumentationName = "io.openliberty.microprofile.telemetry";
+
+    @Inject
+    Config config;
+
+    //See https://github.com/open-telemetry/opentelemetry-java-docs/blob/main/otlp/src/main/java/io/opentelemetry/example/otlp/ExampleConfiguration.java
+    @ApplicationScoped
+    @Produces
+    public OpenTelemetry getOpenTelemetry() {
+
+        HashMap<String,String> telemetryProperties = getTelemetryProperties();
+
+        SpanExporter exporter = getSpanExporter(telemetryProperties);
+
+        Resource serviceNameResource = getServiceName(telemetryProperties);
+        
+        SdkTracerProvider tracerProvider =
+            SdkTracerProvider.builder()
+                .addSpanProcessor(SimpleSpanProcessor.create(exporter))
+                .setResource(Resource.getDefault().merge(serviceNameResource))
+                .build();
+
+        OpenTelemetrySdk openTelemetry =
+            OpenTelemetrySdk.builder().setTracerProvider(tracerProvider).build();
+
+        Runtime.getRuntime().addShutdownHook(new Thread(tracerProvider::close));
+
+        return openTelemetry;
+    }
+
+    @ApplicationScoped
+    private Resource getServiceName(Map<String,String> oTelConfigs){
+        String serviceName = "open-liberty-service";
+        if(oTelConfigs.get("otel.service.name") != null || oTelConfigs.get("OTEL_SERVICE_NAME") != null){
+            serviceName = oTelConfigs.get("otel.service.name");
+        }
+        return Resource.create(Attributes.of(ResourceAttributes.SERVICE_NAME, serviceName));
+    }
+    @ApplicationScoped
+    private SpanExporter getSpanExporter(Map<String,String> oTelConfigs) {
+
+        //default endpoint
+        String endpoint = "http://localhost:14250";
+        String timeout = "10000";
+        if(oTelConfigs.get("otel.traces.exporter") != null){
+            if(oTelConfigs.get("otel.traces.exporter").equals("jaeger") || oTelConfigs.get("OTEL_TRACES_EXPORTER").equals("jaeger")){
+                if(oTelConfigs.get("otel.exporter.jaeger.endpoint") != null){
```

Comment:
This is the wrong order.  Environment variables should have higher priority than system properties according to https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#configuring-the-agent